src/explorer/editors/FileEditor.ts

```diff
@@ -39,13 +44,10 @@ export class FileEditor extends BaseEditor<IAzureNode<FileTreeItem>> {
 
     public async updateData(node: IAzureNode<FileTreeItem>): Promise<string> {
         const webAppClient = nodeUtils.getWebSiteClient(node);
-        const publishingCredential = await node.treeItem.siteWrapper.getWebAppPublishCredential(webAppClient);
-        const kuduClient = new KuduClient(node.treeItem.siteWrapper.appName, publishingCredential.publishingUserName, publishingCredential.publishingPassword);
-
-        const localPath: string = vscode.window.activeTextEditor.document.fileName;
+        const kuduClient: KuduClient = await node.treeItem.siteWrapper.getKuduClient(webAppClient);
+        const localFile: Readable = fs.createReadStream(vscode.window.activeTextEditor.document.uri.fsPath);
         const destPath: string = node.treeItem.path;
-
-        await kuduClient.uploadFile(localPath, destPath);
+        await kuduClient.vfs.putItem(localFile, destPath, { customHeaders: { ['If-Match']: node.treeItem.etag } });

```

Comment:
Just curious - did we ignore the etag stuff before this change?

Reply:
The old API ignored the etag by default.